### PR TITLE
Tests for CustomerList Component

### PR DESCRIPTION
**Pull request by adityanaag3**
### What does this PR do?
Adds tests for CustomerList Component

## The PR fulfills these requirements:

[x] Tests for the proposed changes have been added/updated.
[x] Code linting and formatting was performed.

### CODE REVIEW

File: es-space-mgmt/main/default/lwc/customerList/customerList.html [hunk 1]
```diff
@@ -6,16 +6,18 @@
...
-                <li class="slds-item" key={record.Id}>
-                    <c-customer-tile
-                        customer={record}
-                        object={sobject}
-                        oncustomerselect={publishSelect}
-                    ></c-customer-tile>
-                </li>
-            </template>
-        </ul>
+        <template if:true={customers}>
```
Comment (pozil) [inline es-space-mgmt/main/default/lwc/customerList/customerList.html]: Why did you modify the template?

File: es-space-mgmt/main/default/lwc/customerList/__tests__/data/getCustomerList.json [hunk 2]
```diff
@@ -0,0 +1,26 @@
+[
```
Comment (pozil) [inline es-space-mgmt/main/default/lwc/customerList/__tests__/data/getCustomerList.json:1]: Update your test data to use proper field names with capitals:
```
Id, Email, Name, City, State, Status
```

File: es-space-mgmt/main/default/lwc/customerList/__tests__/customerList.test.js [hunk 3]
```diff
@@ -0,0 +1,179 @@
...
+
+        // Emit data from @wire
+        getCustomerListAdapter.emit(mockCustomerList);
+
+        return Promise.resolve().then(() => {
+            const customerTileElements = element.shadowRoot.querySelectorAll(
+                'c-customer-tile'
+            );
+            expect(customerTileElements.length).toBe(mockCustomerList.length);
+            expect(customerTileElements[0].customer.name).toBe(
```
Comment (pozil) [inline es-space-mgmt/main/default/lwc/customerList/__tests__/customerList.test.js]: General comments: use standard syntax for field names (Capital case on first letter: `Name`). You'll need to update your test data as well.

Comment (pozil) [reply es-space-mgmt/main/default/lwc/customerList/__tests__/customerList.test.js]: Replace that check with
```js
expect(customerTileElements[0].customer).toStrictEqual(mockCustomerList[0]);
```

File: es-space-mgmt/main/default/lwc/customerList/__tests__/customerList.test.js [hunk 4]
```diff
@@ -0,0 +1,179 @@
...
+
+        const element = createElement('c-customer-list', {
+            is: CustomerList
+        });
+
+        element.sobject = SOBJECT_TYPE;
+        document.body.appendChild(element);
+
+        // Emit error from @wire
+        getCustomerListAdapter.error();
```
Comment (pozil) [inline es-space-mgmt/main/default/lwc/customerList/__tests__/customerList.test.js]: Instead of relying on the default error values from the test adapter, pass a custom error here.

**Review by pozil — CHANGES_REQUESTED** [review]

File: es-space-mgmt/main/default/lwc/customerList/customerList.html (hunk 1 above)
Comment (adityanaag3) [reply es-space-mgmt/main/default/lwc/customerList/customerList.html]: Because if `customers` is null (which it is, during page load, before the wire returns data), there is warning that says cannot iterate over undefined... So added this conditional rendering block

File: es-space-mgmt/main/default/lwc/customerList/__tests__/data/getCustomerList.json (hunk 2 above)
Comment (adityanaag3) [reply es-space-mgmt/main/default/lwc/customerList/__tests__/data/getCustomerList.json:1]: The wrapper class returns data with lower case field names, and the component also refers to these fields using lower case only.. So if I change the test data, the tests will fail.

File: es-space-mgmt/main/default/lwc/customerList/__tests__/customerList.test.js (hunk 4 above)
Comment (adityanaag3) [reply es-space-mgmt/main/default/lwc/customerList/__tests__/customerList.test.js]: Done

File: es-space-mgmt/main/default/lwc/customerList/__tests__/customerList.test.js (hunk 3 above)
Comment (adityanaag3) [reply es-space-mgmt/main/default/lwc/customerList/__tests__/customerList.test.js]: Comment 2 is done.. Comment 1: same comment as above

File: es-space-mgmt/main/default/lwc/customerList/customerList.html (hunk 1 above)
Comment (pozil) [reply es-space-mgmt/main/default/lwc/customerList/customerList.html]: Ok so instead of modifying the template which would require extra testing, lets intialize `customers` with an empty array.

File: es-space-mgmt/main/default/lwc/customerList/__tests__/data/getCustomerList.json (hunk 2 above)
Comment (pozil) [reply es-space-mgmt/main/default/lwc/customerList/__tests__/data/getCustomerList.json:1]: Damn, I really thought that these were record fields. Good catch.

**Review by pozil — CHANGES_REQUESTED** [review]
Just one small change and we're good to go.

**Review by pozil — APPROVED** [review]
